stackrox/stackrox
# Replace requests via sagas with requests from System Configuration in UI

## Description

* Dispatch action to update public configuration
* Also delete user telemetry from UI because backend removed in #1865

### Changed files

1. Delete Containers/SystemConfig/ConfigTelemetryDetailWidget.tsx

2. Edit Containers/SystemConfig/Details.tsx
    * Delete user telemetry!

3. Edit Containers/SystemConfig/Page.tsx
    * Add fetch loading and error states to conditional rendering
    * Replace `mapStateToProps` with `fetchSystemConfig` call
    * Replace `mapDispatchToProps` with `saveSystemConfig` call in `SystemConfigForm` also provide `setSystemConfig` and `setIsNotEditing` callbacks as props

4. Edit Containers/SystemConfig/SystemConfigForm/SystemConfigForm.tsx
    * Given response from successful save, call `dispatch` for publicConfig in Redux store and call `setSystemConfig` for detail
    * Delete user telemetry!
    * Add save error state to conditional rendering
    * Add `isDisabled={isSubmitting}` prop to **Cancel** button

5. Edit reducers/index.js
    * Delete `telemetryConfig` and `telemetryConfigSelectors`

6. Edit reducers/systemConfig.js
    * Delete types, actions, reducers, selectors for complete system config
    * Keep types, actions, reducers, selectors for public config

7. Delete reducers/telemetryConfig.js

8. Edit sagas/index.js
    * Delete `telemetryConfig`

9. Edit sagas/systemConfig.js
    * Delete functions for complete system config
    * Keep function for public config

10. Delete sagas/telemetryConfig.js

11. Edit services/SystemConfigService.ts
    * Move wrapped `{ config: { … }}` from function argument as called by sagas to request argument (although the code remained correct when #1733 declared the argument type as unwrapped when it was still wrapped, no compiler error because the function call was from JavaScript)
    * Replace `Promise<{ response: SystemConfig }>` with `Promise<SystemConfig>` for `fetchSystemConfig` function

12. Delete services/TelemetryConfigService.ts

13. Delete types/telemetry.proto.ts

## Checklist

## Testing Performed

### manual testing

1. `yarn build` in ui
2. `yarn start` in ui

    * Visit /main/dashboard: see GET /v1/config/public

    * Visit /main/systemconfiguration: see GET /v1/config
        ![detail](https://user-images.githubusercontent.com/11862657/172693080-dbd7ee4a-e329-458f-8067-e1f29cb3e501.png)

    * Click **Edit** change **Images no longer deployed** and **Expired vulnerability requests**
        ![form-change](https://user-images.githubusercontent.com/11862657/172693156-f154c7aa-ad78-4d63-853c-38f807f059b2.png)

    * Click **Save**: see error message before change to `saveSystemConfig` function argument from wrapped `{ config: { … }}` to `config` itself
        ![Save-errorMessage](https://user-images.githubusercontent.com/11862657/172693174-18f9c573-20de-4416-a89d-991ea10a4f6f.png)

    * Click **Save**: set change in detail view and also PUT /v1/config

        payload has strings for changed number values
        ![Save-payload](https://user-images.githubusercontent.com/11862657/172693203-6745c357-7b5b-4028-a936-28286d06de77.png)

        preview of response had changed number values
        ![Save-preview](https://user-images.githubusercontent.com/11862657/172693226-2aebe5b8-e593-437b-a24b-f72af5da1296.png)

    * Click **Cancel** with or without changes: see original values in detail view

    * Click **Save** after change to **Header** configuration: see PUT /v1/config and header in detail view
        ![Header-enabled](https://user-images.githubusercontent.com/11862657/172693591-c16ca4df-074e-4137-ada6-3072cf217066.png)

    * Click **Save** after disable **Header**: see PUT /v1/config and no header in detail view
        ![Header-disabled](https://user-images.githubusercontent.com/11862657/172693329-fe139273-fec0-4800-b1cf-c8524689f3d7.png)

### integration testing

1. `yarn cypress-open` in ui/apps/platform
    * systemconfig.test.js
